wlauto/instrumentation/poller/poller.c

```diff
@@ -0,0 +1,58 @@
+#include <fcntl.h>
+#include <stdio.h>
+#include <sys/poll.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <errno.h>
+#include <signal.h>
+#include <string.h>
+#include <stdlib.h>
+
+volatile sig_atomic_t done = 0;
+void term(int signum)
+{
+    done = 1;
+}
+
+// argv[1] = Sampling interval in uS
+// argv[2+] = Files to poll
+int main(int argc, char ** argv) {
+    char buf[1024];
+    memset(buf, 0, sizeof(buf));
+    struct timeval current_time;
+    double time_float;
+   
+    //Print headers and open files to poll  
+    printf("time");
+    int filesToPoll[argc - 2];
+    int i;
+    for (i = 0; i < argc-2; i++)
+    {
+        filesToPoll[i] = open(argv[i+2], O_RDONLY);
+	printf(",%s", argv[i+2]);
+    }
+    printf("\n");    
+
+    //Setup SIGTERM handler
+    struct sigaction action;
+    memset(&action, 0, sizeof(struct sigaction));
+    action.sa_handler = term;
+    sigaction(SIGTERM, &action, NULL);
+
+    //Poll files 
+    while (!done) {
+        gettimeofday(&current_time, NULL); 
+       	time_float = (double)current_time.tv_sec;
+        time_float += ((double)current_time.tv_usec)/1000/1000;
+        printf("%f", time_float);
+	for (i = 0; i < argc-2; i++) {
+            read(filesToPoll[i], buf, 1024);
+            lseek(filesToPoll[i], 0, SEEK_SET);
+            buf[strcspn(buf, "\n")] = 0; //Removes "\n"
+	    printf(",%s", buf);
+        }
+	printf("\n");
+        usleep((useconds_t)atoi(argv[1]));
+    }
```

Comment:
You should close the files you've opened (this will also make sure that everything you've written has been flushed).